layers/stateless/sl_ray_tracing.cpp

```diff
@@ -888,306 +896,234 @@ bool StatelessValidation::manual_PreCallValidateGetDeviceAccelerationStructureCo
-                             "is VK_ACCELERATION_STRUCTURE_TYPE_TOP_LEVEL_KHR, but geometryCount is %" PRIu32 ".",
-                             pInfos[i].geometryCount);
-        }
-        if (pInfos[i].type == VK_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL_KHR &&
-            pInfos[i].geometryCount > phys_dev_ext_props.acc_structure_props.maxGeometryCount) {
-            skip |= LogError("VUID-VkAccelerationStructureBuildGeometryInfoKHR-type-03793", objlist, info_loc.dot(Field::type),
-                             "is VK_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL_KHR but geometryCount (%" PRIu32
-                             ") is greater than maxGeometryCount (%" PRIu64 ").",
-                             pInfos[i].geometryCount, phys_dev_ext_props.acc_structure_props.maxGeometryCount);
-        }
-
-        const VkAccelerationStructureGeometryKHR *pGeometry = pInfos[i].pGeometries    ? pInfos[i].pGeometries
-                                                              : pInfos[i].ppGeometries ? pInfos[i].ppGeometries[0]
-                                                                                       : nullptr;
-        if (pGeometry) {
-            if (total_primitive_count > phys_dev_ext_props.acc_structure_props.maxPrimitiveCount) {
-                switch (pGeometry->geometryType) {
-                    case VK_GEOMETRY_TYPE_TRIANGLES_KHR:
-                        skip |= LogError("VUID-VkAccelerationStructureBuildGeometryInfoKHR-type-03795", objlist, info_loc,
-                                         "total number of triangles in all geometries (%" PRIu64
-                                         ") is superior to maxPrimitiveCount "
-                                         "(%" PRIu64 ")",
-                                         total_primitive_count, phys_dev_ext_props.acc_structure_props.maxPrimitiveCount);
-                        break;
-                    case VK_GEOMETRY_TYPE_AABBS_KHR:
-                        skip |= LogError("VUID-VkAccelerationStructureBuildGeometryInfoKHR-type-03794", objlist, info_loc,
-                                         "total number of AABBs in all geometries (%" PRIu64
-                                         ") is superior to maxPrimitiveCount "
-                                         "(%" PRIu64 ")",
-                                         total_primitive_count, phys_dev_ext_props.acc_structure_props.maxPrimitiveCount);
-                        break;
-                    default:
-                        break;
-                }
-            }
+
+    if (check_total_primitive_count) {
+        if (total_primitive_count.total_triangles_count > phys_dev_ext_props.acc_structure_props.maxPrimitiveCount) {
+            skip |= LogError("VUID-VkAccelerationStructureBuildGeometryInfoKHR-type-03795", handle, info_loc,
+                             "total number of triangles in all geometries (%" PRIu64
+                             ") is superior to maxPrimitiveCount "
```

Comment:
```suggestion
                             ") is larger than maxPrimitiveCount "
```

is what I think we use for other limits